client/src/lib/urls.js

```diff
@@ -0,0 +1,7 @@
+export const joinLinks = (...links) => {
```

Comment:
The name is strange as it's not joining links, but parts of a link/URL. Perhaps `joinUrlParts`? or `joinLinkParts` or make it actually a universal helper in style of implode that takes parts and the glue?

Could be something like https://locutus.io/php/strings/implode/ (simplified) or just used as a dependency.

Reply:
Fair enough, changing to `joinUrlParts = (...urlParts) => {` will suffice.

Reply:
Done, except I used `paths` instead of `parts` because I think it's more correct (it doesn't distinguish between, say, query strings or schemes etc so you shouldn't use it to join _the different parts of a URL together_), and because I already made that change before your last comment :p